packages/packages/documents/src/store/__tests__/sync.test.ts

```diff
@@ -0,0 +1,282 @@
+import { createSlice } from '../store';
+import { ExtendedWindow, Slice } from '../../types';
+import { flushListeners } from '@elementor/v1-adapters';
+import { createStore, deleteStore, SliceState, Store } from '@elementor/store';
+import {
+	dispatchCommandAfter,
+	dispatchCommandBefore,
+	dispatchV1ReadyEvent,
+	makeDocumentsManager,
+	makeMockDocument,
+} from './test-utils';
+
+type WindowWithV1Loading = Omit<ExtendedWindow, 'elementor'> & {
+	__elementorEditorV1LoadingPromise?: Promise<void>;
+	elementor?: ExtendedWindow['elementor'];
+}
+
+describe( '@elementor/documents/store/sync', () => {
+	let store: Store<SliceState<Slice>>;
+	let slice: Slice;
+	let extendedWindow: WindowWithV1Loading;
+
+	beforeEach( () => {
+		slice = createSlice();
+		store = createStore();
+
+		syncStore( slice );
+
+		extendedWindow = ( window as unknown as WindowWithV1Loading );
+		extendedWindow.__elementorEditorV1LoadingPromise = Promise.resolve();
+	} );
+
+	afterEach( () => {
+		delete extendedWindow.__elementorEditorV1LoadingPromise;
+		delete extendedWindow.elementor;
+
+		flushListeners();
+		deleteStore();
+	} );
+
+	it( 'should sync documents on V1 load', () => {
+		// Arrange.
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [
+				makeMockDocument( 1 ),
+				makeMockDocument( 2 ),
+			] ),
+		};
+
+		// Act.
+		dispatchV1ReadyEvent();
+
+		// Assert.
+		const storeState = store.getState();
+
+		expect( storeState.documents.documents ).toEqual( {
+			1: {
+				id: 1,
+				title: 'Document 1',
+				status: 'publish',
+				isModified: false,
+				isSaving: false,
+				isSavingDraft: false,
+				userCan: {
+					publish: true,
+				},
+			},
+			2: {
+				id: 2,
+				title: 'Document 2',
+				status: 'publish',
+				isModified: false,
+				isSaving: false,
+				isSavingDraft: false,
+				userCan: {
+					publish: true,
+				},
+			},
+		} );
+	} );
+
+	it.each( [
+		{
+			type: 'V1 load',
+			dispatchEvent: () => dispatchV1ReadyEvent(),
+		},
+		{
+			type: 'document open',
+			dispatchEvent: () => dispatchCommandAfter( 'editor/documents/open' ),
+		},
+	] )( 'should sync current document on $type', ( { dispatchEvent } ) => {
+		// Arrange.
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [
+				makeMockDocument( 1 ),
+				makeMockDocument( 2 ),
+			], 2 ),
+		};
+
+		// Act.
+		dispatchEvent();
+
+		// Assert.
+		const storeState = store.getState();
+		const { currentDocumentId } = storeState.documents;
+		const currentDocument = storeState.documents.documents[ currentDocumentId ];
+
+		expect( currentDocumentId ).toBe( 2 );
+		expect( currentDocument ).toEqual( {
+			id: 2,
+			title: 'Document 2',
+			status: 'publish',
+			isModified: false,
+			isSaving: false,
+			isSavingDraft: false,
+			userCan: {
+				publish: true,
+			},
+		} );
+	} );
+
+	it( 'should sync saving state of a document on V1 load', () => {
+		// Arrange.
+		const mockDocument = makeMockDocument();
+
+		extendedWindow.elementor = {
```

Comment:
Explain? a bit hard to understand what you do.

Reply:
Done